doc/design/quic-programming-guide.md

```diff
@@ -0,0 +1,317 @@
+- Go to `src` dir, and run `gn args out/release-x64`, in the prompted file, input:
+
+```
+is_clang = false
+enable_libaom = true
+rtc_use_h264 = true
+rtc_use_h265 = true
+is_component_build = false
+use_lld = false
+rtc_build_examples = false
+treat_warnings_as_errors = false
+target_cpu = "x64"
+is_debug = false
+ffmpeg_branding = "Chrome"
+rtc_include_tests = false
+owt_include_tests = false
+owt_use_quic = true
+owt_quic_header_root = "d:\\workspace\\quic\\include"; #example
+owt_quic_lib_root = "d:\\workspace\\quic\\bin\\Release" #example
+````
+
+Please be noted the owt\_quic\_header\_root points to the "include" dir the package built from C++ QUIC SDK, and the owt\_quic\_lib\_root points to the bin/release/ or bin/debug directory that contains owt\_quic\_transport.lib|dll|so.
+
+Still in `src` dir, run `ninja -C out/release-x64`. The built binary will be under `out/release-x64/obj/talk/owt/owt.lib`.
+
+#### Linux
+
+You would need Ubuntu 18.04 to build the client SDK. **CentOS is currently not supported**.
+
+- First make sure you have correct proxy set for apt and also for https\_proxy/http\_proxy set for shell if you're behind firewall.
+- Make sure you have python2.7 installed and it's the first python in the path to be used in shell.
+- Run gclient sync. It make take a long time to download a large amount of data. The sync may fail due to some of the dependencies missing. In that case, run src/build/install-build-deps.sh to install the biuld dependencies for Linux.
+- After gclient sync finishes successfully, in `src` dir, run `python scripts/build_linux.py --gn_gen --sdk --use_gcc –arch x64 –scheme release –quic_root=<path_to_quic_sdk_binary_containing_include_and_lib_dir>`
+
+The output library will be under `src/out/release-x64/obj/talk/owt/libowt.a`, and header files that can be used by sample will be under `src/talk/owt/sdk/include/cpp`.
+
+# OWT Native Conferene SDK API
+
+OWT conference SDK provides a series of API for wrapping the underlying QUIC SDK to communicate with the OWT conference server.
+
+Typical calling flow for publishing QUIC stream:
+
+![plot](./pics/publishing_flow.png)
+
+Typical calling flow for Subscribing QUIC stream:
+
+![plot](./pics/subscribing_flow.png)
+
+Please see the conference sample application for more detailed usage.
+
+# OWT QUIC Conference Server
+
+In this section we brief introduce the OWT QUIC Conference Server Build steps.
+
+## How to install development dependencies
+
+In the repository root, use one of following commands to install the dependencies.
+
+- Interactive mode: scripts/installDeps.sh
+- Non-interactive mode: scripts/installDepsUnattended.sh In interactive mode, you need type &quot;yes&quot; to continue installation several times and in non-interactive, the installation continues automatically.
+
+## How to Build OWT Conference Server 
+
+### Requirements
+
+The OWT QUIC conference server can be built on the following platforms:
+
+- Ubuntu 18.04
+- CentOS 7.6
```

Comment:
Please confirm with QA that they tested it on CentOS.

Reply:
Will rely on MCU README for it, as we're not directly put the instruction here.